frame.go

```diff
@@ -1556,8 +1578,18 @@ func (f *framer) writeQueryParams(opts *queryParams) {
 	}
 
 	if opts.keyspace != "" {
+		if f.proto < protoVersion5 {
+			panic(fmt.Errorf("the keyspace can only be set with protocol 5 or higher"))
+		}
 		f.writeString(opts.keyspace)
 	}
+
+	if opts.nowInSeconds != nil {
+		if f.proto < protoVersion5 {
+			panic(fmt.Errorf("now_in_seconds can only be set with protocol 5 or higher"))
+		}
+		f.writeInt(int32(*opts.nowInSeconds))
```

Comment:
NIT: it would make a bit more sense to do these checks and panics earlier in the function when the header flags are being added (similar to how it was before).

Same for the batch write function

Reply:
Very good catch! Somehow it didn't occur to me... I added a little fix for this one.

We probably should do the same for other fields.